### ISSUE TITLE
Unable to create Presto table using minio backend

### ISSUE THREAD

**Issue by dscottcs** [minio/minio#5195]
## Expected Behavior
[Presto](https://prestodb.io/) create 'external' tables using files in S3 in supported formats (e.g. parquet). One table generally corresponds to more than one file in S3, which are contained in an X3 'directory'.  You specify the 'directory' path when creating the table.  What happens in S3 should be transparent to the user.

## Current Behavior
To do this, Presto first sends a HEAD request to the S3 'directory' path to verify that it is a valid path.  Since this path is a directory and not a file, minio rejects this request as invalid and returns a 403 error.  Here is an example, from minio stdout:

```
ERRO[0234] Unable to fetch object info.                  cause=Prefix access is denied: solarscope-raw-data/import/datastore/load/REVENUE/PS3/UTILITYAPIPS37539REV_20171115233257_c source=[object-handlers.go:247:objectAPIHandlers.HeadObjectHandler()] stack=fs-v1-helpers.go:225:fsStatFile fs-v1.go:488:fsObjects.getObjectInfo fs-v1.go:522:fsObjects.GetObjectInfo <autogenerated>:1:(*fsObjects).GetObjectInfo object-handlers.go:245:objectAPIHandlers.HeadObjectHandler api-router.go:44:(objectAPIHandlers).HeadObjectHandler-fm
```

## Possible Solution
Simply allow HEAD request to proceed without error on a non-leaf node.

## Steps to Reproduce (for bugs)
1. Install presto using minio as S3 backend
2. Upload more than one file in a supported format, using the same 'directory' path for all files
3. Create a table in presto using the 'directory' as external location.

## Context
This issue is preventing us from using minio as an S3 replacement in testing.

## Your Environment
We see this issue using dockerized minio obtained from docker hub as well as most recent minio binary running on Ubuntu xenial 16.04LTE

```
vagrant@minio:~$ uname -a
Linux minio 4.4.0-51-generic #72-Ubuntu SMP Thu Nov 24 18:29:54 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux

```

**Comment by harshavardhana** [comment]
This should be fixed in the master branch @dscottcs fixed by PR #5179

**Comment by dscottcs** [comment]
Awesome.  I'll test this and close ticket if we're good.

**Comment by harshavardhana** [comment]
Oh wait its not fixed in master yet, but we have a fix - https://github.com/minio/minio/pull/5179

**Comment by harshavardhana** [comment]
Fixed by #5179
